src/streaming/controllers/MediaController.js

```diff
@@ -570,23 +571,66 @@ function MediaController() {
         let result = [];
         let tmp;
 
-        trackArr.forEach(function (track) {
-            const sum = track.bitrateList.reduce(function (acc, obj) {
-                const resolution = Math.max(1, obj.width * obj.height);
-                const efficiency = obj.bandwidth / resolution;
-                return acc + efficiency;
-            }, 0);
-            tmp = sum / track.bitrateList.length;
-
-            if (tmp < min) {
-                min = tmp;
-                result = [track];
-            } else if (tmp === min) {
-                result.push(track);
-            }
-        });
+        if ( trackArr[0] && (trackArr[0].type === Constants.VIDEO) ) {
```

Comment:
Please split this logic into two separate functions one for audio and one for video which are then called here

Reply:
Done